### PR TITLE
Making hope an addon

### LINKED ISSUE #2: Make and addon from the hope branch
We currently use this branch to flash the prototype on phone:https://github.com/mozilla-b2g/gaia/compare/master...etiennesegonzac:hope?expand=1

The main part is in `apzinit.js`, we need to make it a real b2g addon.

### PR DESCRIPTION
**Pull request by etiennesegonzac**
Fixes #2 .

It's based on top of #10 , I'll wait for this one to land before asking for review here.

### CODE REVIEW

**Comment by etiennesegonzac** [comment]
Ready for review @wilsonpage :)

The CSS variables are defined in the system app.
I can't see any way to make this meaningfully independent from the system app.

Once the scope is a bit more defined, we might be able to land more explicit hooks in the system app to make this better though.

**Comment by jonathanKingston** [comment]
:+1: works for me... not that I am involved in the project just interested :smile:.

Worth having `npm install` in the `make addon`?

**Comment by etiennesegonzac** [comment]
@jonathanKingston cool! yes `npm install` is required, I'll add it to the readme. Thanks!

File: lib/hope-tab-list.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /*global fxosComponent, HopeTab, scheduler */
 (function() {
 
-var debug = 1 ? (...args) => console.log('[hope-tab-list]', ...args) : () => {};
+var debug = 0 ? (...args) => console.log('[hope-tab-list]', ...args) : () => {};
```
Comment (wilsonpage) [inline lib/hope-tab-list.js:4]: Ooops forgot about this, thanks :)

File: Makefile [hunk 2]
```diff
@@ -4,4 +4,7 @@ clean:
 
 .PHONY: profile
 profile:
-	zip -r -9 profile/webapps/system.gaiamobile.org/application.zip assets bower_components css index.html js manifest.webapp
+	zip -r -9 profile/webapps/system.gaiamobile.org/application.zip assets index.html lib node_modules js manifest.webapp
+
+addon:
+	zip addon.zip addon.js manifest.json assets index.html lib node_modules
```
Comment (wilsonpage) [inline Makefile:10]: I'm confused why we need to manually zip this thing up. Doesn't WebIDE do this work for us?

Comment (etiennesegonzac) [inline Makefile:10]: Don't think so... The add-on support is pretty sparse :/

Comment (wilsonpage) [inline Makefile:10]: I don't know much about add-ons, but at MozFest we had a session where people were flashing their own add-ons to devices. i don't remember there being a zip stage.

Comment (etiennesegonzac) [inline Makefile:10]: Not equipped to test it right now but happy to remove it if not needed :)
Was MozFest before or after the change of format?

Comment (wilsonpage) [inline Makefile:10]: @arcturus will know :)

**Comment by wilsonpage** [comment]
r+
- Hoping we can kill the Makefile as a followup :)

File: Makefile (hunk 2 above)
Comment (jonathanKingston) [inline Makefile:10]: I have never zipped an addon in webIDE unless it's required for b2d perhaps?

**Comment by etiennesegonzac** [comment]
File #13

**Pull request merged**